applications/OptimizationApplication/python_scripts/execution_policies/kratos_analysis_execution_policy.py

```diff
@@ -0,0 +1,105 @@
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.component_data_view import ComponentDataView
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem) -> ExecutionPolicy:
+    if not parameters.Has("name"):
+        raise RuntimeError(f"KratosAnalysisExecutionPolicy instantiation requires a \"name\" in parameters [ parameters = {parameters}].")
+    if not parameters.Has("settings"):
+        raise RuntimeError(f"KratosAnalysisExecutionPolicy instantiation requires a \"settings\" in parameters [ parameters = {parameters}].")
+
+    return KratosAnalysisExecutionPolicy(parameters["name"].GetString(), model, parameters["settings"], optimization_problem)
+
+class KratosAnalysisExecutionPolicy(ExecutionPolicy):
+    def __init__(self, name: str, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        super().__init__(name)
+
+        default_settings = Kratos.Parameters("""{
+            "model_part_names" : [],
+            "analysis_module"  : "KratosMultiphysics",
+            "analysis_type"    : "",
+            "analysis_settings": {}
+        }""")
+        self.model = model
+        self.parameters = parameters
+        self.optimization_problem = optimization_problem
+        self.parameters.ValidateAndAssignDefaults(default_settings)
+
+        analysis_module = parameters["analysis_module"].GetString()
+        analysis_type = parameters["analysis_type"].GetString()
+        analysis_settings = parameters["analysis_settings"]
+
+        if analysis_module == "KratosMultiphysics":
+            analysis_module = GetClassModuleFromKratos(analysis_type)
+
+        self.model_parts: 'list[Kratos.ModelPart]' = []
+        analysis_full_module = f"{analysis_module}.{Kratos.StringUtilities.ConvertCamelCaseToSnakeCase(analysis_type)}"
+        self.analysis: AnalysisStage = getattr(import_module(analysis_full_module), analysis_type)(self.model, analysis_settings.Clone())
+
+        if analysis_settings["output_processes"].Has("vtk_output"):
+           self._GetOutputData(analysis_settings["output_processes"]["vtk_output"][0]["Parameters"])
+        elif analysis_settings["output_processes"].Has("gid_output"):
+           self._GetOutputData(analysis_settings["output_processes"]["gid_output"][0]["Parameters"])
+
+    def _GetOutputData(self,output_process_params):
+        self.nodal_solution_step_data_variables = output_process_params["nodal_solution_step_data_variables"]
+        self.nodal_data_value_variables = output_process_params["nodal_data_value_variables"]
+        self.element_data_value_variables = output_process_params["element_data_value_variables"]
+        self.condition_data_value_variables = output_process_params["condition_data_value_variables"]
```

Comment:
This is gonna break in following three cases:
1. The parameters group `vtk_output` and `gid_output` are merely arbitrary names, user can have any name in there, hence this will break if a user specifies a different name.
2. This does not work with the `HDF5`.
3. This does not work with `vtu` output for primal analysis.

Hence, I would remove this mechanism fully, and have something like the following:
```json
{
            "model_part_names" : [],
            "analysis_module"  : "KratosMultiphysics",
            "analysis_type"    : "",
            "analysis_settings": {}
            "analysis_output_settings": {
                     "nodal_solution_step_data_variables": [],
                     "nodal_data_value_variables": [],
                     "element_data_value_variables": [],
                     "condition_data_value_variables": []
            }
}
```

Then we can remove the check also which this process checks whether a vtu output requests this. This is way cleaner.